pkg/gitops_test.go

```diff
@@ -1558,20 +1546,14 @@ func TestGenerateAndPush(t *testing.T) {
 	repo := "http://github.com/testing/testing.git"
 	outputPath := "/fake/path"
 	component := gitopsv1alpha1.Component{
-		Spec: gitopsv1alpha1.ComponentSpec{
-			ContainerImage: "testimage:latest",
-			Source: gitopsv1alpha1.ComponentSource{
-				ComponentSourceUnion: gitopsv1alpha1.ComponentSourceUnion{
-					GitSource: &gitopsv1alpha1.GitSource{
-						URL: repo,
-					},
-				},
-			},
-			TargetPort: 5000,
+		ContainerImage: "testimage:latest",
+		Source: &gitopsv1alpha1.ComponentSource{
+			URL: repo,
```

Comment:
Add coverage for ComponentSource optional fields

Reply:
Removing those unused fields.